models/facet.js

```diff
@@ -132,7 +134,15 @@ Facet.loadIntoCache = function(cache, facetName, allConfigFiles, cb) {
         debug('loading [%s] dataSource into cache', dataSourceName);
         DataSourceDefinition.addToCache(cache, def);
       });
-      
+      cb();
+    });
+  }
+
+  if(middlewares) {
+    steps.push(function(cb) {
+      middlewares.load(cb);
+    }, function(cb) {
+      Middleware.deserialize(cache, facetName, middlewares);
```

Comment:
I looked up the methods I was referring to: `ModelDefinition.getConfigFromCache`, `WorkspaceEntity.getConfigFromData`, `WorkspaceEntity.getDataFromConfig`. If I read the code in this patch correctly, your new serialize/deserialize methods call these `get*From*` method under the hood :+1: 

I agree it's a good idea to keep this logic out of facet.js.

Consider this comment as resolved then.